.github/workflows/integration_test.yaml

```diff
@@ -27,14 +27,30 @@ jobs:
       - name: Setup operator environment
         uses: charmed-kubernetes/actions-operator@main
         with:
-          provider: microk8s
-      - name: Enable microk8s registry
-        # Add a step to wait for the registry to become available after it has
-        # been deployed to avoid any issues with trying to push images to it
-        # before it is ready
+          provider: lxd
+      - name: Install microk8s
         run: |
-          sudo microk8s.enable registry
+          sudo snap install microk8s --classic
+          sudo microk8s.status --wait-ready
+          sudo microk8s.enable dns hostpath-storage registry
+          sudo usermod -a -G microk8s $USER
+          sudo chown -f -R $USER ~/.kube
+      - name: Ensure microk8s is ready
+        run: |
+          sudo microk8s.kubectl -n kube-system rollout status -w deployment/hostpath-provisioner
+          sudo microk8s.kubectl -n kube-system rollout status -w deployment/coredns
           sudo microk8s.kubectl -n container-registry rollout status -w deployment/registry
+      - name: Bootstrap microk8s
+        run: |
+          sudo microk8s config > ${HOME}/.kube/config
+          sg microk8s -c "juju bootstrap microk8s k8s-ctrl"
+          juju switch k8s-ctrl
+      # [20220824] charmed-kubernetes/actions-operator@main does not work on
+      # ubuntu-22.04, but charm runs on 22.04 which has Python 3.10, hence
+      # installing Python 3.10 here
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
```

Comment:
We have a standard for which Python version to use, it is based on the Ubuntu series the charm runs: https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#charm-ubuntu-and-python-version

It looks like this was copied from another repo (my guess is the Jenkins agent k8s operator) where this step was necessary because the `charmed-kubernetes/actions-operator@main` action did not support jammy at the time (I believe it does now) so the workflow needed to run on focal whilst the charm runs on jammy